pkgs/checks/lib/src/checks.dart

```diff
@@ -97,7 +94,7 @@ Subject<T> check<T>(T value, {String? because}) => Subject._(_TestContext._root(
 ///
 /// Asynchronous expectations are not allowed in [condition] and will cause a
 /// runtime error if they are used.
-CheckFailure? softCheck<T>(T value, Condition<T> condition) {
+CheckFailure? softCheck<T>(T value, void Function(Subject<T>) condition) {
```

Comment:
The function type here shows up a lot, I would consider a typedef (or two)

Reply:
Done